src/zjs_callbacks.c

```diff
@@ -315,6 +334,20 @@ void zjs_remove_callback(int32_t id)
     }
 }
 
+void zjs_signal_callback_args(int32_t id, void* args, uint32_t size)
+{
+    if (args && size) {
+        int ret = zjs_port_ring_buf_put(&ring_buffer, (uint16_t)id, 0, (uint32_t*)args, (uint8_t)size / 4);
```

Comment:
No, that's not what I said. Zephyr *can* tell you the size of the next item in the ring buffer. If you pass it a '0' for the size, and there were arguments stored, it will tell you your buffer is not big enough and return you the size you need. So it's a bit inconvenient, but it already does this.

Having the set size like 4 or 5 that I talked about is not inefficient. That space is not being wasted inside the ring buffer. It is only when you go to read it out, you have your waiting buffer be that size so that it can handle any possible output. (Like having a 256 char buf to receive a string even though most of the time the string will be 10-15 chars.)

However, you did point out that in some cases like the timer functions the number of arguments is defined by the app writer, not by us. So there we have two options: either a configurable MAX_ARGS that they need to increase in Zephyr if they're going to use more... or we malloc this read buffer when our default is not big enough.